src/Stratis.Bitcoin.Features.Wallet/WalletTransactionHandler.cs

```diff
@@ -372,18 +381,71 @@ protected virtual void AddRecipients(TransactionBuildContext context)
             if (context.Recipients.Any(a => a.Amount == Money.Zero))
                 throw new WalletException("No amount specified.");
 
-            if (context.Recipients.Any(a => a.SubtractFeeFromAmount))
-                throw new NotImplementedException("Substracting the fee from the recipient is not supported yet.");
+            int totalRecipients = context.Recipients.Count(r => r.SubtractFeeFromAmount);
+
+            // If we have any recipients that require a fee to be subtracted from the amount, then
+            // calculate fee and evenly distribute it among all recipients. Any remaining fee should be
+            // subtracted from the first recipient.
+            if (totalRecipients > 0 && context.TransactionFee != null)
+            {
+                Money fee = context.TransactionFee;
+                long recipientFee = fee.Satoshi / totalRecipients;
+                long remainingFee = fee.Satoshi % totalRecipients;
+
+                for (int i = 0; i < context.Recipients.Count; i++)
+                {
+                    Recipient recipient = context.Recipients[i];
+
+                    if (recipient.SubtractFeeFromAmount)
+                    {
+                        // First receiver pays the remainder not divisible by output count.
+                        long feeToSubtract = i == 0 ? remainingFee + recipientFee : recipientFee;
+                        long remainingAmount = recipient.Amount.Satoshi - feeToSubtract;
+                        if (remainingAmount <= 0)
+                            throw new WalletException($"Fee {feeToSubtract} is higher than amount {recipient.Amount.Satoshi} to send.");
+
+                        recipient.Amount = new Money(remainingAmount);
+                    }
 
-            foreach (Recipient recipient in context.Recipients)
-                context.TransactionBuilder.Send(recipient.ScriptPubKey, recipient.Amount);
```

Comment:
based on my comment, if it's true that TransactionBuilder.SubtractFees does what I expect it to do, it would be enough to call context.TransactionBuilder.SubtractFees after this line

Reply:
@MithrilMan I checked that code but I don't think it is evenly distributing tx fees across all recipients.

It just subtracts total fee from the send. Can you check that code, it is hard to follow

Reply:
you may be right, at a first sight it just subtract it from the last output created before the call, maybe @NicolasDorier could shed light on this, it seems transaction builder doesn't supports deducting fees from multiple outputs

I would expect a way to deduct fee from specific outputs and then be able to pay the eventual remaining fees at the end
like `context.TransactionBuilder.Send(recipient.ScriptPubKey, recipient.Amount, deductFee);` where deductFee is a boolean

Reply:
Check this commit where I tried to rewrite it in a clearer way https://github.com/MetacoSA/NBitcoin/pull/763/commits/83b049d6f80d65910776b4c816040ae235c7d569

The basic idea is: I try to find the builder with the Target is affected by the SubstractFee.
I remove the builder from the list, and create the same one but sending less to the output. If there is not enough money remaining in the output (dust), I delete the builder completely.

Reply:
> like context.TransactionBuilder.Send(recipient.ScriptPubKey, recipient.Amount, deductFee); where deductFee is a boolean

I did not wanted to touch anything that was working before so the syntax is in NBitcoin:

```csharp
builder
.Send(1BTC, alice)
.SubstractFee();
```
The call to `SubstractFee` just keep in memory the last builder above it.

Reply:
> The basic idea is: I try to find the builder with the Target is affected by the SubstractFee.
I remove the builder from the list and create the same one but sending less to the output. If there is not enough money remaining in the output (dust), I delete the builder completely.

I think this is somewhat similar to what I am doing in this PR, i.e. find a send builder and change it with a new amount. A fee is evenly distributed between recipients who have subtract fee flag set and any remaining fee is subtracted from first recipient as per bitcoin client.